src/GitHub.Exports/Services/VSServices.cs

```diff
@@ -86,11 +89,68 @@ public string GetActiveRepoPath()
             return repo?.Info?.Path ?? string.Empty;
         }
 
+        public IEnumerable<ISimpleRepositoryModel> GetKnownRepositories()
+        {
+            try
+            {
+                return PokeTheRegistryForRepositoryList();
+            }
+            catch (Exception ex)
+            {
+                VsOutputLogger.WriteLine(string.Format(CultureInfo.CurrentCulture, "Error loading the repository list from the registry '{0}'", ex));
+                return Enumerable.Empty<ISimpleRepositoryModel>();
+            }
+        }
+
+        const string TEGitKey = @"Software\Microsoft\VisualStudio\14.0\TeamFoundation\GitSourceControl";
+        static RegistryKey OpenGitKey(string path)
+        {
+            return Registry.CurrentUser.OpenSubKey(TEGitKey + "\\" + path, true);
+        }
+
+        static IEnumerable<ISimpleRepositoryModel> PokeTheRegistryForRepositoryList()
+        {
+            using (var key = OpenGitKey("Repositories"))
+            {
+                return key.GetSubKeyNames().Select(x =>
+                {
+                    using (var subkey = key.OpenSubKey(x))
+                    {
+                        var path = subkey.GetValue("Path") as string;
```

Comment:
Potential `NullReferenceException` here. If we want to throw an exception when `subkey` is null (since the caller is handling it), we probably should throw something better and more informative than `NullReferenceException`.